ui/src/file_upload.tsx

```diff
@@ -113,13 +113,17 @@ export const
       maxFileSizeBytes = max_file_size ? convertMegabytesToBytes(max_file_size) : 0,
       maxSizeBytes = max_size ? convertMegabytesToBytes(max_size) : 0,
       fileExtensions = file_extensions ? file_extensions.map(e => e.startsWith('.') ? e : `.${e}`) : null,
+      rejectXhr = React.useRef<(reason?: any) => void>(),
+      xhrRef = React.useRef<XMLHttpRequest>(),
       upload = async (uploadFiles = files) => {
         const formData = new FormData()
         uploadFiles.forEach((f: File) => formData.append('files', f))
 
         try {
           const { responseText } = await new Promise<XMLHttpRequest>((resolve, reject) => {
-            const xhr = new XMLHttpRequest()
+            xhrRef.current = new XMLHttpRequest()
+            rejectXhr.current = reject
```

Comment:
We can take advantage the power of JS closures and get rid of `xhrRef` as well.

```suggestion
            rejectXhr.current = () => {
               reject()
               xhr.abort()
               setPercentComplete(0)
            }
```